test/integration/granting_permissions_test.rb

```diff
@@ -125,39 +92,6 @@ class GrantingPermissionsTest < ActionDispatch::IntegrationTest
       assert_not_includes @user.permissions_for(app), "never"
     end
 
-    should "support granting app-specific permissions on apps with greater than eight supported permissions" do
-      app = create(
-        :application,
-        name: "MyApp",
-        with_supported_permissions: %w[pre-existing removing adding never-1 never-2 never-3 never-4 never-5 never-6],
-      )
-      @user.grant_application_signin_permission(app)
-      @user.grant_application_permissions(app, %w[pre-existing removing])
-
-      visit edit_user_path(@user)
-      click_on "Manage permissions"
-      click_on "Update permissions for MyApp"
-      uncheck "removing"
-      click_button "Update permissions"
-      assert_includes @user.permissions_for(app), "pre-existing"
-      %w[removing adding never-1 never-2 never-3 never-4 never-5 never-6].each do |permission|
-        assert_not_includes @user.permissions_for(app), permission
-      end
-
-      click_on "Update permissions for MyApp"
-      click_button "Add permission"
-      flash = find("div[role='alert']")
-      assert flash.has_content?("You must select a permission.")
-
-      select "adding"
-      click_button "Add permission"
-      assert_includes @user.permissions_for(app), "pre-existing"
-      assert_includes @user.permissions_for(app), "adding"
-      %w[removing never-1 never-2 never-3 never-4 never-5 never-6].each do |permission|
-        assert_not_includes @user.permissions_for(app), permission
-      end
-    end
-
```

Comment:
is this test exactly the same as the one you've removed on lines 43 - 75?

Reply:
and indeed 236 - 269?

Reply:
Yup, only differences are:
- the type of admin we're testing with. Given all these test the same actions, the other tests feel sufficient for testing authorisation/privileges. These 9+ permissions tests are more about presentation and user interaction so I feel like they only really need one test per function, rather than one per function per admin type
- the third one seems to grant the admin sign in permissions for the app, though I think that line might not be doing anything useful: `@super_org_admin.grant_application_signin_permission(app)`